content/integrations/awslambda.md

```diff
@@ -21,27 +21,93 @@ If you haven't already, set up the [Amazon Web Services integration first](/inte
 In the Amazon Web Services integration tile, ensure that Lambda is checked under metric collection.
 
 To send custom metrics to Datadog, you must print a log line from your Lambda, using the following format:
-`MONITORING|unix_epoch_timestamp|value|metric_type|my.metric.name|#tag1:value,tag2`
 
-Please ensure the `unix_epoch_timestamp` is in seconds (not milliseconds).
+~~~
+MONITORING|unix_epoch_timestamp|value|metric_type|metric.name|#tag1:value,tag2
+~~~
+
+## Notes on each section
+
+### MONITORING
+* This is used to find the log statement within Cloudwatch Logs and pull it into Datadog
+
+### unix_epoch_timestamp
+* Please ensure the `unix_epoch_timestamp` is in seconds (not milliseconds)
+
+### value
+* The value of the metric **must** be a number.
+* For metric_type `check` the options are, `'0': OK, '1': WARNING, '2': CRITICAL, '3': UNKNOWN`
+
+### metric_type
+* Supported metric types are `count, gauge, histogram, and check`
+
+### metric.name
+* A unique name to identify your metric
 
-For example, here is sample snippet for printing a valid custom metric, from your Lambda function (in Python):
+### tags
+* Optionally add tags to apply to this metric
+* If you do not want to add any tags use the format: `MONITORING|unix_epoch_timestamp|value|metric_type|metric.name`
+* The tag `function_name` will automatically be applied to custom metrics
+
+
+## Sample snippets (in Python):
+
+### Count/Gauge
 
 ~~~
 unix_epoch_timestamp = int(time.time())
 value = 42
 metric_type = 'count'
+#metric_type = 'gauge'
 metric_name = 'my.metric.name'
 tags = ['tag1:value', 'tag2']
+
 print('MONITORING|{0}|{1}|{2}|{3}|#{4}'.format(
     unix_epoch_timestamp, value, metric_type, metric_name, ','.join(tags)
 ))
 ~~~
 
-Note: This integration requires the AWS permissions `logs:DescribeLogGroups`, `logs:DescribeLogStreams`, and `logs:FilterLogEvents` to be fully enabled. Also, counts and gauges are the only metrics types currently supported.
+### Histogram
+
+~~~
+unix_epoch_timestamp = int(time.time())
+metric_type = 'histogram'
+metric_name = 'my.metric.name.hist'
+tags = ['tag1:value', 'tag2']
+
+for i in xrange(0,10):
+	print('MONITORING|{0}|{1}|{2}|{3}|#{4}'.format(
+    	unix_epoch_timestamp, i, metric_type, metric_name, ','.join(tags)
+))
+~~~
+
+Note: Using the histogram metric type provides `avg, count, max, min, 95p, and median` values. These values are calculated at one second granularity. 
+
+### Service Check
+
+~~~
+unix_epoch_timestamp = int(time.time())
+value = 1
+metric_type = 'check'
+metric_name = 'my.metric.name.check'
+
+print('MONITORING|{0}|{1}|{2}|{3}'.format(
+	timestamp, value, metric_type, metric_name
+))
+~~~
+
+
+## Required Permssions
```

Comment:
This feels a little lost beneath all the code examples. Can we put it near the beginning of **Configuration** section?

(After all, it's needed for non-custom metrics and custom metrics alike, yes?)

Also, a short sentence or two of explanation would be nice, rather than just a header and then a list.